### ISSUE TITLE
Group: Calendar issue

### ISSUE THREAD

**Issue by MattiasJay** [Skjutsgruppen/skjutsgruppen-reactnative#489]
When opening group calender and then pressing a ride (blue or pink ball in calender) the list starts at 2014 and not on the current date. Also trie scrolling in the list and you see that design overlaps in an ugly way (iOS).

**Comment by shresthaKiran** [comment]
Referenced in #519

**Comment by MattiasJay** [comment]
This is still wrong @shresthaKiran and is not referenced in issue 519 what I can see. Still a problem with opening 2014 instead of current date.

**Comment by ruchidhami** [comment]
@MattiasJay  I have recheck what you have asked for, but it works perfectly . Can you explain little bit more?

**Comment by MattiasJay** [comment]
The calendar in groups still have issues @anjesh @ruchidhami @anjhero : 

- [x] 1. When trying to open the calendar in the group Sthlm - GBG the whole app crashes. IOS, iPhone 6. This group: https://skjutsgruppen.nu/g/149

- [x] 2. Rides that are added through the menu (the "plus" and the offered/asked and then shared to a group are still not in the calendar. We have talked about this being important since that is a common user flow. Look at the ride @saugatmaharjan shared to the group "Chobhar" on December 6th and 5th (scroll in the group "Chobhar" to see them). Then click on the calendar in the group "Chobhar", they are not in the calendar.

- [x] 3. It still is a problem with calender dates opening the wrong date, that this issue thread started with (se above). To see this: Go to the Chobhar group -> press calender -> Go to December 2018 -> press December 19. What happens is that it shows the first entry of the group in June instead of scrolling to December 19. We have talked about that it should scroll. 
This is even more apparent in groups like "Sthlm - GBG" where it starts on 2014 instead of the actual date pressed. 
To clarify, see the video about issue 489 that I've uploaded to Slack.

- [x] 4. Grey background. Just like all onlays the calender should generate a grey-ish transparent background behind it when opened.

- [x] 5. Design for when both offered and asked for ride is on the same day is missing. It should be a split blue and pink. See the design here: https://projects.invisionapp.com/d/main#/console/12489615/265439270/preview

- [x] 6. The calender is not rounded on top and does not have the correct spacing/design as the design here: https://projects.invisionapp.com/d/main#/console/12489615/265439270/preview

**Comment by ruchidhami** [comment]
@MattiasJay @anjhero  
1. When trying to open the calendar in the group Sthlm - GBG the whole app crashes. IOS, iPhone 6 
    > **We had also faced this issue earlier and fixed it before the weekend start. Can you please confirm it.** 
2.  Rides that are added through the menu (the "plus" and the offered/asked and then shared to a group are still not in the calendar. We have talked about this being important since that is a common user flow. Look at the ride @saugatmaharjan shared to the group "Chobhar" on December 6th and 5th (scroll in the group "Chobhar" to see them). Then click on the calendar in the group "Chobhar", they are not in the calendar.
    > **Since the trips of December 6th and 5th hasn't been created through the group "Chobhar". It was shared after it was created . There is only five trips in "Chobhar Group"**
3. Design for when both offered and asked for ride is on the same day is missing. It should be a split blue and pink.  
   > **Split ball has been implemented previously. Can you please confirm it?**

**Comment by MattiasJay** [comment]
Hi again @ruchidhami !

Yes, **number 1** -the crashing- is confirmed yesterday and today in the latest downloadable version of the App in App Store. It still crashes.

Regarding **number 2**; yes this is what I meant. IF participant adds a ride from menu -> "+" -> offer a ride /and/or/ ask for a ride AND share to selected groups THEN the rides(s) should appear in the selected groups calendars. This is because this is a very common user flow. I just tested this with the Chobhar group, and it didn't work.

Regarding number 3: I now tested with adding rides only from the actual group, and yes the split ball then occured. Great! Good job @ruchidhami !

**Comment by ruchidhami** [comment]
1. When trying to open the calendar in the group Sthlm - GBG the whole app crashes. IOS, iPhone 6. 
> Strange . I had tested it on Iphone 6 and as well in android , it works fine. But definitely , I would look over it.
2.  Rides that are added through the menu (the "plus" and the offered/asked and then shared to a group are still not in the calendar. We have talked about this being important since that is a common user flow. Look at the ride @saugatmaharjan shared to the group "Chobhar" on December 6th and 5th (scroll in the group "Chobhar" to see them). Then click on the calendar in the group "Chobhar", they are not in the calendar.
>  Share a trip on a different groups doesn't linked that trip on shared groups. A group may have various trips but a trip only have one group , which was implemented in our application. @anjesh @anjhero  @MattiasJay  If we do so ,then their would be a huge changes in a database schema and as well as in the functionality logic.
3. It still is a problem with calender dates opening the wrong date, that this issue thread started with (se above). To see this: Go to the Chobhar group -> press calender -> Go to December 2018 -> press December 19. What happens is that it shows the first entry of the group in June instead of scrolling to December 19. We have talked about that it should scroll.
This is even more apparent in groups like "Sthlm - GBG" where it starts on 2014 instead of the actual date pressed.
> I got this issue. This will be fixed soon .
4. Grey background. Just like all onlays the calender should generate a grey-ish transparent background behind it when opened.
> @MattiasJay  Can you please confirm it from the below screenshot ? 
5.  The calendar is not rounded on top and does not have the correct spacing/design as the design here: https://projects.invisionapp.com/d/main#/console/12489615/265439270/preview
> This can be done as per design , So if we implement this, calendar should be dragged as either vertically or horizontally to select dates. What would you prefer? @MattiasJay  Can you also please confirm the spacing and rounded on top in calendar from the below screenshot? 
   ![screenshot_skjutsgruppen_20190104-112806 1](https://user-images.githubusercontent.com/13533382/50675242-e5a99f80-1014-11e9-8b23-31c50549a101.png)
